### PR TITLE
fix: [GO Feature Flag server] Implement ProviderStatus

### LINKED ISSUE #489: [go-feature-flag] doesn't implement 'status'
The Go-Feature-Flag provider doesn't implement status: https://github.com/open-feature/js-sdk-contrib/blob/main/libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts

In order for it to stay functional after [this](https://github.com/open-feature/js-sdk/pull/507) is released, it needs to be updated to set its `status` field (it should start with it `NOT_READY` and update it to `READY` once its `bgScheduler` interval is started.

### PR DESCRIPTION
**Pull request by thomaspoignant**
## This PR
- Implement ProviderStatus

Fixes #489

### CODE REVIEW

File: libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts [hunk 1]
```diff
@@ -68,6 +69,9 @@ export class GoFeatureFlagProvider implements Provider {
   // logger is the Open Feature logger to use
   private logger?: Logger;
 
+  // status of the provider
```
Comment (beeme1mr) [inline libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts:72]: It's up to you, but I would remove this comment. I think the code is self-explanatory.

**Review by beeme1mr — APPROVED** [review]
Approved with some minor feedback.

File: libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts (hunk 1 above)
Comment (thomaspoignant) [inline libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts:72]:
```suggestion
```

**Review by toddbaert — COMMENTED** [review]
lgtm as well

**Comment by liran2000** [comment]
Minor comment, according to the [spec](https://openfeature.dev/specification/sections/providers#requirement-245), I think this indication is missing:
```
Requirement 2.4.5
The provider SHOULD indicate an error if flag resolution is attempted before the provider is ready.
```

**Comment by thomaspoignant** [comment]
@liran2000 you are totally right, I've open this PR https://github.com/open-feature/js-sdk-contrib/pull/505 to fix that.

**Comment by toddbaert** [comment]
> Minor comment, according to the [spec](https://openfeature.dev/specification/sections/providers#requirement-245), I think this indication is missing:
> 
> ```
> Requirement 2.4.5
> The provider SHOULD indicate an error if flag resolution is attempted before the provider is ready.
> ```

Nice catch @liran2000 . We actually missed this in flagd-js as well. I created and issue: https://github.com/open-feature/js-sdk-contrib/issues/517

**Pull request merged**